includes/class-wc-rest-authentication.php

```diff
@@ -66,6 +67,99 @@ protected function is_request_to_rest_api() {
 		return apply_filters( 'woocommerce_rest_is_request_to_rest_api', $woocommerce || $third_party );
 	}
 
+	/**
+	 * Check if this is a request to WCCOM Site REST API.
+	 *
+	 * @return bool
+	 */
+	protected function is_request_to_wccom_site_rest_api() {
+		if ( empty( $_SERVER['REQUEST_URI'] ) ) {
+			return false;
+		}
+
+		$rest_prefix = trailingslashit( rest_get_url_prefix() );
+		$request_uri = esc_url_raw( wp_unslash( $_SERVER['REQUEST_URI'] ) );
+
+		return false !== strpos( $request_uri, $rest_prefix . 'wccom-site/' );
+	}
+
+	/**
+	 * Verify WooCommerce.com request from a given body and signature request.
+	 *
+	 * @param string $body                Request body.
+	 * @param string $signature           Request signature found in
+	 *                                    X-Woo-Signature header.
+	 * @param string $access_token_secret Access token secret for this site.
+	 *
+	 * @return bool
+	 */
+	protected function verify_wccom_request( $body, $signature, $access_token_secret ) {
+		// phpcs:disable
```

Comment:
Please, don't disable all rules, better to only disable a few ones when required.

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/24123/commits/59a5c6d7545ffc2e1553e2bfbf8376e05eb7f00f.